src/Entity/Repository/CourseRepository.php

```diff
@@ -365,20 +427,145 @@ public function findByUserId(
             $sql .= ' OFFSET :offset';
         }
 
-        $query = $this->_em->createNativeQuery($sql, $rsm);
-        $query->setParameter('user_id', $userId);
+        $conn = $this->getEntityManager()->getConnection();
+        $stmt = $conn->prepare($sql);
+        $stmt->bindValue(":user_id", (int) $userId);
         foreach ($params as $field => $label) {
             $value = $criteria[$field];
-            $query->setParameter($label, $value);
+            if (is_array($label)) {
+                $arr = array_combine($label, $value);
+                foreach ($arr as $label => $value) {
+                    $stmt->bindValue($label, $value);
+                }
+            } else {
+                $stmt->bindValue($label, $value);
+            }
         }
 
         if (isset($limit)) {
-            $query->setParameter('limit', (int)$limit);
+            $stmt->bindValue(":limit", (int) $limit);
         }
         if (isset($offset)) {
-            $query->setParameter('offset', (int)$offset);
+            $stmt->bindValue(":offset", (int) $offset);
+        }
+        $stmt->execute();
+        $rows = $stmt->fetchAll();
+        $stmt->closeCursor();
+        return $rows;
```

Comment:
I cry... but I move on....!

Reply:
i know what you mean - this should be `$rhett` instead of `$rows`.